src/main/kotlin/skytils/skytilsmod/features/impl/mining/MiningFeatures.kt

```diff
@@ -553,14 +552,19 @@ class MiningFeatures {
         var lastTPLoc: BlockPos? = null
         var waypoints = HashMap<String, BlockPos>()
         var deadCount: Int = 0
-        val locationsToInternal = hashMapOf<String, String>(
-            "Lost Precursor City" to "internal_city",
-            "Jungle Temple" to "internal_temple",
-            "Goblin Queen's Den" to "internal_den",
-            "Mines of Divan" to "internal_mines",
-            "Khazad-dûm" to "internal_bal",
-            "Fairy Grotto" to "internal_fairy"
-        )
-        val SBE_DSM_PATTERN: Pattern = Pattern.compile("\\\$(SBECHWP\\b|DSMCHWP):(.*?)@-?\\d*,-?\\d*,-?\\d*")
+        val SBE_DSM_PATTERN: Pattern = Pattern.compile("\\\$(SBECHWP\\b|DSMCHWP):(?<stringLocation>.*?)@(?<x>-?\\d*),(?<y>-?\\d*),(?<z>-?\\d*)")
+
+        /**
+         * Removes everything from the string except for the part that matches the Pattern
+         * @param pattern The Pattern that is singled out
+         */
+        fun String.removeExceptPattern(pattern: Pattern): String {
+            val matcher = pattern.matcher(this)
+            return if (matcher.find()) {
+                matcher.group(0)
+            } else {
+                ""
+            }
+        }
```

Comment:
what's the point of this? With Kotlin's Regex class you can call find and it'll return a nullable matchresult which will be non null when there is a match